### PR TITLE
doc[autograd]: improve autograd plugin readme

### PR DESCRIPTION
**Pull request by yaugenst-flex**
Updating the autograd plugin readme in light of adjoint plugin deprecation (moved the transition guide to the bottom now since adjoint plugin should not be the first thing people see here anymore). Also did a bunch of other formatting and general style/content updates to hopefully make it more digestible and comprehensive at the same time.

## Greptile Summary

Comprehensive update to `tidy3d/plugins/autograd/README.md` to establish autograd as the primary automatic differentiation solution, replacing the deprecated adjoint plugin.

- Reorganized content structure to prioritize core autograd functionality, moving adjoint migration guide to the end
- Added detailed explanations of adjoint method mechanics and updated capabilities section with organized tables
- Enhanced documentation with improved code examples and best practices section
- Improved overall readability while maintaining technical depth

### CODE REVIEW

File: tidy3d/plugins/autograd/README.md [hunk 1]
```diff
@@ -1,254 +1,204 @@
...
+params = anp.array([2.0]) # Initial width
+optimizer = optax.adam(learning_rate=0.01)
+opt_state = optimizer.init(params)
+
+for i in range(20):
+    value, gradient = value_and_grad_fn(params)
+    updates, opt_state = optimizer.update(-gradient, opt_state, params) # Use -gradient to maximize
+    params = optax.apply_updates(params, updates)
+    print(f"Step {i+1}: Value = {value:.4f}, Width = {params[0]:.3f}")
+```
```
[the remark this answers is not in the document]
Comment (tylerflex) [reply tidy3d/plugins/autograd/README.md:77]: good bot. we should probably mention optax somewhere as our "preferred" optimizer. I ran into this with the metalens notebook because I realized that if we introduce optax halfway through these notebooks, it could be annoying to install and come back to it.

File: tidy3d/plugins/autograd/README.md [hunk 2]
```diff
@@ -1,254 +1,204 @@
...
-A good starting point to get familiar with them is the [autograd tutorial](https://github.com/HIPS/autograd/blob/master/docs/tutorial.md).
+### Differentiable Parameters (Simulation Inputs)
 
-Some of the most important autograd "Don'ts" are:
+| Component Type | Traceable Attributes | Example Use Case |
+| :--- | :--- | :--- |
+| **Geometry** | | |
+| `Box` | `.center`, `.size` | Shape Optimization |
+| `Cylinder` | `.center`, `.radius`, `.length` | Shape Optimization |
+| `PolySlab` | `.vertices`, `.slab_bounds`, `dilation` | Shape Optimization |
```
[the remark this answers is not in the document]
Comment (tylerflex) [reply tidy3d/plugins/autograd/README.md:90]: good bot

**Review by tylerflex — CHANGES_REQUESTED** [review]
some comments from the human reviewer (me):

1. the intro makes a lot of mentions of the previous jax plugin. since we're removing it for 2.9, I suggest we just frame this document as if someone was starting from scratch, what is adjoint, autodiff, and very basic how does it work with python functions. We can perhaps remove this and add to the "migration guide" at the end  the end of the document (where we also link to the deprecated adjoint plugin docs?) Otherwise it might be confusing to mix up with the main point of the document.

2. I actually think "basic workflow" should come before "how it works". it will be helpful to get right to the minimal code example I think. otherwise the text is a bit confusing to follow and people might give up.

3. Where do I find out more about what's available in `plugins.autograd`? would be good to link?

4. Perhaps some links to the specific notebook(s) that demonstrate(s) some of the specific features listed could be useful if we want to include it. Or at very least, a link to the inverse design examples page.

5. A few sharp bits I feel are worth mentioning.
   a. what is `.values`, `.data` and when do I need to use them?
   b. `.item()` is useful (previously I was doing `np.sum(x.values)`. We should mention it?
   c. Can I autograd through Job, Batch? what about things like web.upload()?

6. As far as adjoint plugin deprecation is concerned, maybe in a separate PR, we should probably unlink the deprecated adjoint docs from the main docs TOC. We can keep them around and link to them here but no need for most users to be able to navigate there from the main Docs?

File: tidy3d/plugins/autograd/README.md [hunk 3]
```diff
@@ -1,254 +1,204 @@
...
+    )
+    return sim
+
+# 2. Objective function returning a scalar
+def objective_fn(width):
+    sim = make_simulation(width)
+    sim_data = td.web.run(sim, task_name="optimization_step")
+    # Objective: maximize power in the fundamental mode
+    mode_amps = sim_data["monitor_name"].amps.sel(direction="+", mode_index=0)
+    return anp.sum(anp.abs(mode_amps.values)**2)
```
Comment (groberts-flex) [inline tidy3d/plugins/autograd/README.md:62]: should this be .data instead of .values?

File: tidy3d/plugins/autograd/README.md [hunk 4]
```diff
@@ -1,254 +1,204 @@
...
+
+# 3. Get the value and gradient function
+value_and_grad_fn = autograd.value_and_grad(objective_fn)
```
Comment (groberts-flex) [inline tidy3d/plugins/autograd/README.md:65]: should we show the example with the internal tidy3d value_and_grad instead of autograd's version?

**Review by groberts-flex — COMMENTED** [review]
looks good, left a few small comments/questions

**Comment by yaugenst-flex** [comment]
closing in favor of https://github.com/flexcompute/tidy3d/pull/2999
